src/Shared/Components/ComponentDescriptorInstance.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components
+{
+    // The DTO that we data-protect and include into any
+    // generated component marker and that allows the client
+    // to bootstrap a blazor server-side application.
+    internal struct ComponentDescriptorInstance
```

Comment:
`ComponentDescriptorInstance` -> `ComponentMarker`?

Reply:
I renamed this and all its friends to ServerComponent and ServerComponentSerializer

Reply:
![image](https://user-images.githubusercontent.com/6995051/63101325-ae498c80-bf2d-11e9-867e-6c1f1f476790.png)

I've also reduced the shared code just to the DTOs and shared settings
![image](https://user-images.githubusercontent.com/6995051/63101361-c0c3c600-bf2d-11e9-92bf-3c580fc5f4f5.png)

Reply:
In fact, I split the classes in two and renamed ServerComponentSerializer to ServerComponentSerializationSettings and it just contains the jsonoptions and the dataprotection purpose